### PR TITLE
Added move type

### LINKED ISSUE #10: Add Move type
We need a type to represent the metadata of a Pokemon's moves at runtime. This won't be responsible for handling side effects of moves yet.

See: https://github.com/StevensSEC/pokemonbattlelib/wiki/Requirements#moves

### Tasks

- [ ] Define the contents of `type Move struct` that are necessary to have in a Pokemon battle, or to be able to obtain external metadata.
- [ ] Implement the `Stringer` interface for `Move`. Example: https://tour.golang.org/methods/17

### PR DESCRIPTION
**Pull request by Jordan883**
closes #10

### CODE REVIEW

File: move.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	Type string
+	Category string
+	Max_PP int
+	Priority int
+	Power int
+	Accuracy int
+	//Side effects not implemented yet
+}
+
+//Stringer implementation
```
Comment (dyc3) [inline move.go:17]: (nit) obvious, should be omitted

File: move.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+func (m Move) String() string{
+	return fmt.Sprintf("Name: %v\nType: %v\nCategory: %v\nMax PP: %v\nPriority: %v\nPower: %v\nAccuracy: %v\n", m.Name, m.Type, m.Category, m.Max_PP, m.Priority, m.Power, m.Accuracy);
```
Comment (dyc3) [inline move.go:19]: I feel like this is way too much information.

File: move.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package pokemonbattlelib
+
+import "fmt"
+
+//A type for moves
```
Comment (dyc3) [inline move.go:5]: Needs to be slightly more descriptive
```suggestion
// Represents a Pokemon's move. Moves can deal damage, heal the user or allies, or cause status effects.
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: move.go (hunk 2 above)
Comment (Jordan883) [reply move.go]: Changed string to name, type, power, and acc in shorter format

File: move.go (hunk 1 above)
Comment (Jordan883) [reply move.go]: Removed excess comments

**Review by dyc3 — APPROVED** [review]

**Comment by carpenterd777** [comment]
This implementation has the `Type` field as a `string`. shouldn't it be an `ElementalType`?

**Comment by dyc3** [comment]
That can be changed later, when #8 is done

**Pull request merged**